packages/babel-parser/src/parser/expression.js

```diff
@@ -1204,6 +1204,30 @@ export default class ExpressionParser extends LValParser {
         return this.parseTopicReferenceThenEqualsSign(tt.bitwiseXOR, "^");
       }
 
+      case tt.doubleCaret: {
+        const pipeProposal = this.getPluginOption(
+          "pipelineOperator",
+          "proposal",
+        );
+        const pluginTopicToken = this.getPluginOption(
+          "pipelineOperator",
+          "topicToken",
+        );
+
+        // The `^^` token is valid only when:
+        // the pipe-operator proposal is active,
+        // its "pipeProposal" is configured as "hack",
+        // and "topicToken" is configured as "^^".
+        // If the pipe-operator proposal settles on a token that is not ^^,
+        // then this token type may be removed.
+        if (pipeProposal === "hack" && pluginTopicToken === "^^") {
+          return this.parseTopicReference(pipeProposal);
+        } else {
+          throw this.unexpected();
```

Comment:
We should use `this.expectPlugin(["pipelineOperator", { proposal: "hack", topicToken: "^^" }])`?

Reply:
Is this branch reachable? We only form `doubleCaret` when `hackPipeWithDoubleCaretIsActive` is true.